.lintstagedrc.js

```diff
@@ -0,0 +1,8 @@
+module.exports = {
+  // ESLint's configuration will let it ignore built files
+  // in `dist` or `package` that we want left alone, as well as the polyfills
+  // Not using `lint:js` or `lint:scss` as they ran slower than the raw linter
+  // (maybe because they use their cache while we only lint a handful of files)
+  '*.{cjs,js,mjs}': ['eslint --max-warnings 0 --fix'],
+  '*.scss': ['stylelint --max-warnings 0 --fix']
```

Comment:
Wondering if the config looks expressive enough to set `FILES=` environment variable

Which we could pass into the original npm scripts?

```console
FILES=abc.mjs npm run lint:js
FILES=def.scss npm run lint:scss
```

You'd add defaults to the npm script in **package.json** (scroll right)

```json
"lint:js": "eslint --cache --cache-location .cache/eslint --cache-strategy content --color --ignore-path .gitignore --max-warnings 0 ${FILES:-\"**/*.{cjs,js,mjs}\"}",
"lint:scss": "stylelint --cache --cache-location .cache/stylelint --cache-strategy content --color --ignore-path .gitignore --max-warnings 0 ${FILES:-\"{app,src}/**/*.scss\"}",
```

`${FILES:-\"**/*.{cjs,js,mjs}\"}`
`${FILES:-\"{app,src}/**/*.scss\"}`

Otherwise we're missing these flags making the hook a bit slow (no cache):

```console
--cache --cache-location .cache/eslint --cache-strategy content --color --ignore-path .gitignore
```

Reply:
It's a JavaScript config so you could read the **package.json** and grab the `lint:js`/`lint:scss` scripts?

```js
const pkg = require(join(__dirname, 'package.json'))
```

Take a look at their [JavaScript function example](https://github.com/okonet/lint-staged#example-ignore-files-from-match):

```js
// lint-staged.config.js
import micromatch from 'micromatch'

export default {
  '*.js': (files) => {
    // from `files` filter those _NOT_ matching `*test.js`
    const match = micromatch.not(files, '*test.js')
    return `eslint ${match.join(' ')}`
  },
}
```

Quick string replacement (watch out for escaping) and that does the job?

Reply:
Or revisit a separate "CLI" version of the npm script suggestion from https://github.com/alphagov/govuk-frontend/pull/2848#discussion_r971810349?

```json
"lint:js": "npm run lint:js:cli -- \"**/*.{cjs,js,mjs}\"",
"lint:js:cli": "eslint --cache --cache-location .cache/eslint --cache-strategy content --color --ignore-path .gitignore --max-warnings 0",
"lint:scss": "npm run lint:scss:cli -- \"{app,src}/**/*.scss\"",
"lint:scss:cli": "stylelint --cache --cache-location .cache/stylelint --cache-strategy content --color --ignore-path .gitignore --max-warnings 0"
```

So the hooks can run the CLI version, bypassing the default files arg

```suggestion
  '*.scss': ['npm run lint:scss:cli -- --fix']
```